openhtf/util/measurements.py

```diff
@@ -31,6 +31,18 @@
 themselves are similarly output in the 'measured_values' field on the
 PhaseRecord.  See test_record.py for more information.
 
+Validation of undimensioned measurements happens when they are set, so that
+users of the HTTP API can see PASS/FAIL outcome on those measurements
+immediately after they are set.  Multidimensional measurements, however,
+don't usually make sense to validate until all data is available, so they
+instead enter a PARTIALLY_SET outcome state until the end of the test phase,
+at which point they are validated and become with PASS or FAIL.  Note that
+validators are only called at the end of the phase if at least one value was
```

Comment:
only multidim measurements by virtue of undimensioned measurements getting validated when set, and therefore never getting called at the end of the phase regardless.